src/main/scala/viper/gobra/frontend/Parser.scala

```diff
@@ -464,14 +465,21 @@ object Parser {
         "pure" <~ eos ^^^ PureClause
       }
 
-      functSpecClause.* ~ "pure".? ^^ {
+      val specClauses = if(withTrusted) {
+        functSpecClause | "trusted" <~ eos ^^^ TrustedClause
+      } else {
+        functSpecClause
+      }
```

Comment:
the variable `specClauses` is unnecessary. We can just add `trusted" <~ eos ^^^ TrustedClause` to the definition of `functSpecClause`.

Reply:
In particular, I do not think that we need the `withTrusted` parameter. Instead, we need to pick the parser for the body according to the result of parsing the spec (look at my suggestion above).